src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -58,6 +59,35 @@ public override void Initialize (AnalysisContext context)
 				var invocationOperation = (IInvocationOperation) context.Operation;
 				ProcessInvocationOperation (context, invocationOperation);
 			}, OperationKind.Invocation);
+
+			context.RegisterSymbolAction (context => {
+				ProcessGenericTypes (context);
+			}, SymbolKind.NamedType);
+		}
+
+		static void ProcessGenericTypes (SymbolAnalysisContext context)
+		{
+			// Check the Symbol to see the generics relationship, specifically around DAMT
+			// We will initially explore if an unbounded generic type has mismatched annotations with its base-class,
+			// which is an open generic type. i.e. Bar<T> : Foo<[DAMT] T>
+
+			INamedTypeSymbol derivedType = (INamedTypeSymbol) context.Symbol;
+			var baseType = derivedType.BaseType;
+
+			//Check if generic parameter of derived and generic argument of the base have mismatched annotation
+
+			var typeParams = baseType!.TypeParameters;
+			var typeArgs = baseType.TypeArguments;
+
+			for (int i = 0; i < typeParams.Length; i++) {
+				var sourceValue = new SymbolValue (typeArgs[i]);
+				var targetValue = new SymbolValue (typeParams[i]);
+				// if a source is known, we dont need to warn
+				if (sourceValue.Source.Kind == SymbolKind.NamedType)
```

Comment:
We should still call the method below. Right now it won't do anything, but going forward it will "mark" the type. Which could produce additional diagnostics (if it has RUC methods on it for example).